core/src/main/java/hudson/model/TreeView.java

```diff
@@ -140,9 +140,16 @@ public void deleteView(View view) throws IOException {
     }
 
     public View getView(String name) {
-        for (View v : views)
-            if(v.getViewName().equals(name))
+        for (View v : views) {
+            if (v.getViewName().equals(name))
                 return v;
+            if (v instanceof ViewGroup) {
```

Comment:
I originally implemented it as an extra loop, but later inlined it. However, I forgot about the order in the search results which sounds very convincing for me. I will restore the old implementation.